src/lib.rs

```diff
@@ -40,18 +40,27 @@ pub mod slave;
 pub use self::slave::{Slave, SlaveId};
 
 mod codec;
+
 mod error;
+pub use self::error::ResponseError;
 
 mod frame;
-pub use self::frame::{Address, Exception, FunctionCode, Quantity, Request, Response};
+pub use self::frame::{
+    Address, Exception, ExceptionResponse, FunctionCode, Quantity, Request, Response,
+};
 
-/// Specialized [`std::result::Result`] type for `Modbus` client API.
+/// Specialized [`std::result::Result`] type for type-checked responses of the _Modbus_ client API.
+///
+/// The payload is generic over the response type.
 ///
 /// This [`Result`] type contains 2 layers of errors.
 ///
 /// 1. [`std::io::Error`]: An error occurred while performing I/O operations.
-/// 2. [`Exception`]: An error occurred on the `Modbus` server.
-pub type Result<T> = std::io::Result<std::result::Result<T, Exception>>;
+/// 2. [`ResponseError`]: An error occurred on the _Modbus_ server or when matching the received response with the request.
+pub type Result<T> = std::io::Result<std::result::Result<T, ResponseError>>;
+
+/// Result of a generic [`Request`].
+pub type ResponseResult = Result<Response>;
```

Comment:
```suggestion
type ResponseResult = Result<Response>;
```

For me, it reads much better when standard types are used. An alias is introduced here that I as a user don't recognize as such so quickly.

Reply:
https://github.com/slowtec/tokio-modbus/pull/254/commits/a12143943f7520470dfd2f93494672bafe958733